blocks/inner-blocks/README.md

```diff
@@ -0,0 +1,147 @@
+InnerBlocks
+===========
+
+InnerBlocks exports a pair of components which can be used in block implementations to enable nested block content.
+
+Refer to the [implementation of the Columns block](https://github.com/WordPress/gutenberg/tree/master/blocks/library/columns) as an example resource.
+
+## Usage
+
+In a block's `edit` implementation, simply render `InnerBlocks`, optionally with `layouts` of available nest areas:
+
+Then, in the `save` implementation, render `InnerBlocks.Content`. This will be replaced automatically with the content of the nested blocks.
+
+```jsx
+import { registerBlockType, InnerBlocks } from '@wordpress/blocks';
+
+registerBlockType( 'my-plugin/my-block', {
+	// ...
+
+	edit( { className } ) {
+		return (
+			<div className={ className }>
+				<InnerBlocks />
+			</div>
+		);
+	},
+
+	save() {
+		return (
+			<div>
+				<InnerBlocks.Content />
+			</div>
+		);
+	}
+} );
+```
+
+_Note:_ A block can render at most a single `InnerBlocks` and `InnerBlocks.Content` element in `edit` and `save` respectively. To create distinct arrangements of nested blocks, refer to the `layouts` prop documented below.
+
+_Note:_ Since the save step will automatically apply props to the element returned by `save`, it is important to include the wrapping `div` in the above simple example even though we are applying no props of our own. In a real-world example, you may have your own attributes to apply to the saved markup, or sibling content adjacent to the rendered nested blocks.
+
+## Props
+
+### `InnerBlocks`
+
+#### `layouts`
```

Comment:
Maybe I'm wrong (actually, I'm probably wrong but worth asking :P), so this feels like a way to do two dimension handling of inner blocks, which is IMO specific to the columns blocks. Can't we avoid this entirely and use a private inner block `core/column` for each column. So the columns block would be a list of `core/column` and each `core/column` would contain another list of blocks? What's not possible with this one dimension approach?

Reply:
The thinking is the extra dimension is semantically unnecessary.

Reply:
I would be in favor of the private `core/column` block, in my experience it makes handling the document smoother. I also don't understand how this is all coming together, so I am probably not the best candidate for judgement.